examples/counter/cli/src/main.rs

```diff
@@ -44,85 +38,30 @@ struct Args {
 }
 
 fn main() -> Result<()> {
-    let (tx, rx) = unbounded::<Message>();
-
-    let strong_tx = Arc::new(tx);
-    let tx = Arc::downgrade(&strong_tx);
-
-    let core = Arc::new(Core::new::<Capabilities>());
-
-    // Kick off with the given command
-    main_loop(&core, Args::parse().cmd.into(), tx.clone())?;
-    drop(strong_tx); // tx may still live in a side-effect futures
+    let filter = EnvFilter::try_from_default_env().unwrap_or_else(|_| "info,surf=warn".into());
+    let format = tracing_subscriber::fmt::layer();
+    tracing_subscriber::registry()
+        .with(filter)
+        .with(format)
+        .init();
 
-    // Continue until there's no more work to do
-    while let Ok(msg) = rx.recv() {
-        main_loop(&core, msg, tx.clone())?;
-    }
+    let command = Args::parse().cmd;
 
-    Ok(())
-}
-
-fn main_loop(
-    core: &Arc<Core<Effect, App>>,
-    message: Message,
-    tx: Weak<Sender<Message>>,
-) -> Result<(), ErrReport> {
-    match message {
-        Message::Event(event) => {
-            for effect in core.process_event(event) {
-                process_effect(effect, core, tx.clone())?
-            }
-        }
-        Message::Effect(effect) => process_effect(effect, core, tx)?,
-    }
+    let core = core::new();
+    let event = command.into();
+    let (tx, rx) = unbounded::<Effect>();
 
-    Ok(())
-}
+    core::update(&core, event, &Arc::new(tx))?;
 
-fn process_effect(
-    effect: Effect,
-    core: &Arc<Core<Effect, App>>,
-    tx: Weak<Sender<Message>>,
-) -> Result<(), ErrReport> {
-    match effect {
-        Effect::Render(_) => {
+    while let Ok(effect) = rx.recv() {
```

Comment:
Oh I see why the Cat facts CLI is doing what it's doing... because there _could_ be multiple renders per effect loop settle